engine/src/main/java/org/destinationsol/game/SolObject.java

```diff
@@ -16,41 +16,187 @@
+     * force to object's internal {@link Body}.
+     *
+     * @param force Force to apply to the object.
+     * @param game  Game this object belongs to.
+     * @param acc   Whether this objects is accelerating by the force, such as in gravity.
+     */
     void receiveForce(Vector2 force, SolGame game, boolean acc);
 
+    /**
+     * Returns object's position.
+     * <p>
+     * This method usually just returns the vector retrieved from object's internal {@link Body}.
+     *
+     * @return Position of the object.
+     */
     Vector2 getPosition();
 
+    /**
+     * Called when an object is too far from player/camera, and is thus to be converted to {@link FarObject}.
+     * <p>
+     * This method is used for optimization of game's objects, by converting to their more resource friendly versions
+     * when they get too far from player/camera. The {@code FarObject} returned by this method should be usable to get
+     * copy of this object back in reverse. If null is returned by this method, this object is designated to be removed
+     * when it gets too far from player/camera, and instead of recreating from {@code FarObject}, to be generated again.
+     *
+     * @return FarObject representation of this object, or null if object is to be removed.
+     */
     FarObject toFarObject();
 
+    /**
+     * Returns list of all {@link Drawable Drawables} this object has assigned.
+     *
+     * @return List of drawables this object has.
+     */
     List<Drawable> getDrawables();
 
+    /**
+     * Returns object's angle.
+     * <p>
+     * This method usually just returns the angle retrieved from object's internal {@link Body}.
+     *
+     * @return Angle of this object, in degrees.
+     */
     float getAngle();
 
+    /**
+     * Returns object's speed.
+     * <p>
+     * This method usually just returns the vector retrieved from object's internal {@link Body}.
+     *
+     * @return Speed of this object.
+     */
     Vector2 getSpeed();
 
+    /**
+     * Called whenever this object touches another object, and can be used for custom collisions.
+     * <p>
+     * This method is for instance used for collecting {@link Loot} by ships when these two touch. This method is not to
+     * be used for handling of applying force between the two objects nor playing any collision sounds, unless the
+     * object is specifically created to do so. This method should however apply collision damage. This method can be
+     * freely left blank if there is no custom contact handling.
+     *
+     * @param other      Object this object touches.
+     * @param absImpulse Impulse the objects apply to each other.
+     * @param game       Game this object belongs to.
+     * @param collPos    Position where the two objects touch.
+     */
     void handleContact(SolObject other, float absImpulse, SolGame game, Vector2 collPos);
 
+    /**
+     * Used for retrieval of object's debug string.
```

Comment:
`* Returns a {@link String} containing a well-formatted list of debug information related to the state of the object.`

Side note: The `to` method name prefix conventionally specifies conversion. A good example is `toFarObject`. The usage of the prefix here is invalid. This method should be renamed to something like `dumpDebugInformation` (or better).

Reply:
Actually, this method should return something else than null only when really necessary. Only ships on AIPilots return non-null values from this method now. You can think of this method as of cleaner version of good old System.out.println(). When you need to learn contents of some variable for debugging, you put it in, afterwards, you remove it.

Reply:
While that may be true for the current code-base, I believe that there should _always_ be a way to retrieve debug data on the current state of the object, whether it is needed or not. There should also be a mechanism to easily view that debug information in-game, without having to close it, change code, and recompile the game. This will help immensely for things like user bug reports, and locating the problem in hard to replicate one-off bugs. I say this from personal experience :) This may be something to consider in the future, what do you think? Pinging @vampcat, @Cervator (and any others interested) for opinions.

Reply:
Imagine every single ship, every single bullet, every single shard ofeverything, every single planet tile had 10 lines of text below it. I don't think that will help with debugging much, I rather think that it will cause any debug information you may need to retrieve unreadable

Reply:
That's not how debug information works. There should be mechanisms in place to output debug information just for the object that you want when you want it. From what I can understand, DestSol does not have any such mechanisms in place, but this is still something to discuss in the future. My recommendation for this javadoc has not changed.

Reply:
Javadocs should represent what the method does, not what it should do and be used to in future. For that, I might add a TODO, and I think I will if you do not opose that, possibly with your wording in a non-javadoc comment. I am trying to add TODOs to thinks that would not require much work to be fixed/improved for new contributors, and this possibly seems like the case. Is this option acceptable to you?

Reply:
I completely understand what you are getting at, but I still think we should change the method documentation summary statement.

First, the current summary statement provides no information whatsoever. The method name itself specifies as much information as the summary statement (in that the method returns a debug string) and makes it redundant. 

Furthermore, there are two basic things I expect of **good** debug information:
- It should be well formatted and easy to read.
- It should contain relevant information not easily found elsewhere, especially data which is not displayed in the UI.
Expectations may differ from person to person, but from what I have noticed, these two qualities  are generally considered as absolutely necessary for the debug information to be of any value.

In addition, when documenting _interface_ methods, there is no other option but to document what the method _should_ do. As such, the documentation is often considered as the specifications of the contract defined by the interface. The implementors may choose to deviate from the contract, but they must accept full responsibility. Specifically in this case, if the only non-null implementation in `AiPilot` (as you mentioned) does not return a "well-formatted list of debug information related to the state of the object", it is (according to my definition of ideal debug information) failure on part of the `AiPilot` implementation.

Do the reasons I listed above (specifically the first reason) help change your mind?

Reply:
A bit of spin-off point: What about creating a DebugInfoCollector, to where depositing each objects debug info would be its own responsibility? Like, a single interface for depositing debug info of everything, not just solObjects, that can be completely turned off/on from game options?

Reply:
Catching back up - there was some chat on Slack on this topic :-)

Or rather, there seem to be 2-3 separate topics:

* More advanced debug capacity. Yep, everybody wants this, I believe. Out of scope for this PR but somebody should totally write up an issue. Also I fired up Factorio just to learn what `F4` did ;-) That looks like a great example to follow to allow enabling/disabling certain debug features 👍 
* What to actually do for this specific review comment on this PR right now. I largely agree with @eviltak's words of wisdom, but also hear @Adrijaned's desire to just float along sprinkling javadoc and todos rather than fix all the things :-)
  * The name does seem wrong, but yep that's pretty much out of scope for this PR and should get a nice TODO
  * The summary description (first sentence) - yeah, doesn't really add anything, although other stuff being added does, which is nice. Little tricky what to do with it since this setup may change a lot over the next few months (if anybody starts working on that area / deeper debug stuffs). Evil's example might be a good alternative for now. I'm honestly not sure how much intent we should bake into javadoc like that, especially with changes on the horizon. Maybe such potentially changing intent should be added in a TODO instead? "TODO: Clarify intent, current setup as of this comment is x, may be advisable to change to y in the future" ?

I'd suggest tweaking at the text just a bit, filing a follow-up issue (or two, or three), then moving on along with merging this soonish :-)

Reply:
@Cervator makes a good point --  we really need to begin to wrap things up, we have bigger fish to fry :)

In that regard, @Adrijaned, I request you to speed through this item, which is the last (as of now). I think we all agree there are quite a few future possibilities with the debug system, and we could choose to go in one of many directions. But, as far as I can envision, this method should largely stay the same for most options. Future changes notwithstanding, I really do feel that the summary statement should be changed to something more relevant and descriptive bearing in mind what this method is _expected_ to do. The exact changes I leave to your discretion, @Adrijaned. Feel free to use my proposed statement as a guideline.

Reply:
Somewhat edited/improved, added TODO about debug rework, with link to this PR